### PR TITLE
feat(telemetry): add description and units to standard instruments

### PR DESCRIPTION
**Pull request by bnjjj**
This PR adds description and units to standard instruments available in the router. These descriptions and units have been copy pasted directly from the [OpenTelemetry semantic conventions](https://opentelemetry.io/docs/specs/semconv/http/http-metrics/) and are needed for better integrations with APMs.

---

**Checklist**

Complete the checklist (and note appropriate exceptions) before the PR is marked ready-for-review.

- [x] Changes are compatible[^1]
- [x] Documentation[^2] completed
- [x] Performance impact assessed and acceptable
- Tests added and passing[^3]
    - [x] Unit Tests
    - [x] Integration Tests
    - [x] Manual Tests

**Exceptions**

*Note any exceptions here*

**Notes**

[^1]: It may be appropriate to bring upcoming changes to the attention of other (impacted) groups. Please endeavour to do this before seeking PR approval. The mechanism for doing this will vary considerably, so use your judgement as to how and when to do this.
[^2]: Configuration is an important part of many changes. Where applicable please try to document configuration examples.
[^3]: Tick whichever testing boxes are applicable. If you are adding Manual Tests, please document the manual testing (extensively) in the Exceptions.

### CODE REVIEW

File: apollo-router/src/plugins/telemetry/config_new/instruments.rs [hunk 1]
```diff
@@ -294,7 +319,11 @@ impl InstrumentsConfig {
                             increment: Increment::Custom(None),
                             condition: Condition::True,
                             histogram: Some(
-                                meter.f64_histogram("http.client.response.body.size").init(),
+                                meter
+                                    .f64_histogram("http.client.response.body.size")
+                                    .with_unit(Unit::new("By"))
+                                    .with_description("	Size of HTTP client response bodies.")
```
Comment (BrynCooke) [inline apollo-router/src/plugins/telemetry/config_new/instruments.rs:325]:
```suggestion
                                    .with_description("Size of HTTP client response bodies.")
```

File: apollo-router/src/plugins/telemetry/config_new/instruments.rs [hunk 2]
```diff
@@ -261,7 +282,11 @@ impl InstrumentsConfig {
                             increment: Increment::Custom(None),
                             condition: Condition::True,
                             histogram: Some(
-                                meter.f64_histogram("http.client.request.body.size").init(),
+                                meter
+                                    .f64_histogram("http.client.request.body.size")
+                                    .with_unit(Unit::new("By"))
+                                    .with_description("	Size of HTTP client request bodies.")
```
Comment (BrynCooke) [inline apollo-router/src/plugins/telemetry/config_new/instruments.rs:288]:
```suggestion
                                    .with_description("Size of HTTP client request bodies.")
```

File: apollo-router/src/plugins/telemetry/config_new/instruments.rs [hunk 3]
```diff
@@ -160,7 +170,11 @@ impl InstrumentsConfig {
                             increment: Increment::Custom(None),
                             condition: Condition::True,
                             histogram: Some(
-                                meter.f64_histogram("http.server.response.body.size").init(),
+                                meter
+                                    .f64_histogram("http.server.response.body.size")
+                                    .with_unit(Unit::new("By"))
+                                    .with_description("Size of HTTP server response bodies")
```
Comment (BrynCooke) [inline apollo-router/src/plugins/telemetry/config_new/instruments.rs:176]:
```suggestion
                                    .with_description("Size of HTTP server response bodies.")
```

**Review by BrynCooke — APPROVED** [review]
Approved with suggested fixes for typos.

**Review by garypen — APPROVED** [review]

**Pull request merged**